pkg/destroy/baremetal/baremetal.go

```diff
@@ -41,3 +47,59 @@ func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.
 		Logger:                  logger,
 	}, nil
 }
+
+// deleteStoragePool destroys, deletes and undefines any storagePool left behind during the creation
+// of the bootstrap VM
+func deleteStoragePool(conn *libvirt.Connect, logger logrus.FieldLogger) error {
+	logger.Debug("Deleting baremetal bootstrap volumes")
+
+	pools, err := conn.ListStoragePools()
+	if err != nil {
+		return errors.Wrap(err, "list storage pools")
+	}
+
+	for _, pname := range pools {
+		if !strings.Contains(pname, "-bootstrap") {
```

Comment:
Can we get the cluster name here? I think it would be safer. It's possible to use the same libvirt host for multiple clusters, which means we could potentially delete a storage pool being used by someone else.

Reply:
Added check in the latest update to include cluster name...